language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/packages/PackageConfigSchemaRequest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langserver.extensions.ballerina.packages;
+
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+
+/**
+ * Request format for ConfigSchemaGenerator endpoint.
+ *
+ * @since 2.0.0
+ */
+public class PackageConfigSchemaRequest {
+
+    private TextDocumentIdentifier documentIdentifier;
+
+    protected TextDocumentIdentifier getDocumentIdentifier() {
+        return documentIdentifier;
+    }
+
+    public void setDocumentIdentifier(TextDocumentIdentifier documentIdentifier) {
+        this.documentIdentifier = documentIdentifier == null ? null :
```

Comment:
Do we allow the document identifier to be null? If so what's the usage?

Reply:
Ideally, this should not be null in order to handle the request, but since this is the request model I feel like it should be fine to hold the values coming into it (including null) and perform the null check when serving the request. WDYT?

Reply:
So the better approach would be to use an annotation. So we can enforce the best practice from the API level. Shall we do that? If we do so, there is no need to do a null check when serving the request